cmd/terramate/cli/cli_test.go

```diff
@@ -75,6 +75,63 @@ source = "%s"
 	)
 }
 
+func TestBugModuleMultipleFilesSameDir(t *testing.T) {
+	const (
+		modname1 = "1"
+		modname2 = "2"
+		modname3 = "3"
+	)
+
+	s := sandbox.New(t)
+
+	mod2 := s.CreateModule(modname2)
+	mod2MainTf := mod2.CreateFile("main.tf", "# module 2")
+
+	mod3 := s.CreateModule(modname2)
+	mod3.CreateFile("main.tf", "# module 3")
+
+	mod1 := s.CreateModule(modname1)
+	mod1.CreateFile("main.tf", `
+module "changed" {
+	source = %q
+}
+	`, "../2")
+
+	mod1.CreateFile("secret.tf", `
+module "any" {
+	source = "anything"
+}
+
+module "any2" {
+	source = "anything"
+}
+`)
+
+	stack := s.CreateStack("stack")
+
+	stack.CreateFile("main.tf", `
+module "mod1" {
+    source = %q
+}
+`, stack.ModSource(mod1))
+
+	git := s.Git()
+	git.CommitAll("first commit")
+	git.Push("main")
+	git.CheckoutNew("change-the-module-2")
+
+	mod2MainTf.Write("# changed")
+
+	git.CommitAll("module 2 changed")
+
+	cli := newCLI(t, s.BaseDir())
+	want := stack.RelPath() + "\n"
+	assertRunResult(t, cli.run(
+		"list", s.BaseDir(), "--changed"),
```

Comment:
I think we don't need the basedir as a parameter anymore since we changed the design to allow injecting the working dir. BUt definitely not blocking/essential/etc, just sharing.